backend/routes/rides.js

```diff
@@ -16,21 +17,34 @@ router.get('/', async function (req, res, next) {
     //TODO: validate login state - login user = given userid
     //TODO: search logic for begin and end date
     const query = JSON.parse(req.query['query']);
-    console.log(query);
-    const originCoords = query['originCoords'];
-    const destCoords = query['destCoords'];
-    const beginDate = query['beginDate'];
-    const endDate = query['endDate'];
-    const distance = query['distance']; // within x miles
+    if (DEBUG) {
+        console.log(query);
+    }
+    const {originCoords, destCoords, time, beginDate, endDate, distance} = query;
+    // build date range
+    const rideTime = new Date(time);
+    const dayStart = new Date(rideTime.getFullYear(), rideTime.getMonth(), rideTime.getDate());
+    const dayEnd = new Date(rideTime.getFullYear(), rideTime.getMonth(), rideTime.getDate() + 1);
```

Comment:
Sorry I don't get the reason for doing this here. Why don't you just preprocess these date calculation and pass your dayStart as beginDate and dayEnd as endDate to the query you send from the frontend instead of modifying the interface here? Or do you think there are some advantages of doing this, and we should just stick with this design for GET /api/rides?query=

Reply:
because beginDate and endDate specify a time as well. JS Time objects don't offer an explicit way to check if two dates are on the same day

Reply:
However, this was also based on the fact that the lines for beginDate and endDate were commented out in dbTest.js, and also because this was before I knew about the postman collection lol. I suppose its just a matter if whether we want to do date calculation in frontend or backend. What do you think? I'm not sure about any advantages/disadvantages

Reply:
The naming might be a bit confusing. But here, the endDate has nothing to do with the "return date". So when you do a search, you can search for a range of date e.g. 08/11-08/13 and the query will return all rides that departs within these three days.

Reply:
I prefer doing the calculation in the frontend. But we can also just add support for a search query with only one date&time input and return all rides that departs during that day. @micheyale What do you say? Also I noticed that on Figma, the SEARCH RESULT PAGE from the Low-Fidelity Wireframes is supporting a input like sep9-sep25 for Ride Date.

Reply:
@EvelynL09 oh, Alex is right about only giving the user the option to search a day. We will not be allowing them to search a range of dates because there will be some problem such as user might have different timing preference on different day. The search result page from figma actually shows the round trip. So sep9 will be the date to the destination and sep25 will be the date returning from the destination. For the MVP, we will be implementing the search result page (one way trip) on figma. sorry for the misconception.

Reply:
However, we will still need the beginDate and endDate since we are allowing user to search within a day a range of time with the sort by time function(where we will implement probably next week). So an example search query for the beginDate and endDate with the sort by time function implemented is beginDate: 2020-11-08T01:30:00 and endDate: 2020-11-08T04:00:00.

In terms of the date calculation, I'm thinking having it on the frontend would be better since our GET API is solely focusing on querying from DB and sending the correct HTTP response.

Reply:
@alex-dinh could u migrate the date calculation logic to the frontend and sent the beginDate and endDate as a query to the backend? Since we will not have the filter by time function implemented for MVP, by default the time query should be set to 24 hrs of the searched day. For example, if a user search sep 9, the search query will be beginDate: 2020-09-07T00:00:00 and endDate: 2020-09-07T23:59:00. Is this possible?

Reply:
Done! time querying calculations are now in the frontend